src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -365,7 +365,7 @@ public DebuggedProcess(bool bLaunched, LaunchOptions launchOptions, ISampleEngin
                 }
                 catch (Exception e) when (ExceptionHelper.BeforeCatch(e, Logger, reportOnlyCorrupting: true))
                 {
-                    if (this.ProcessState == MICore.ProcessState.Exited)
+                    if (_terminating)
```

Comment:
You might want `_terminating || ...` but you certainly don't want to **only** do terminate here as the same could happen if the debuggee ran to completion, or detach cases. It might want sense to instead of using '_terminating' use a new '_exitting'.